# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: b22268575af1
# Full Hash: b22268575af1492bb26867e00ba96bf98414212d
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-04-01 09:40:15
# Description:
#   Bug 1956153 - Create actors directly on the main thread r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D243689
# ==============================================================================

diff -r 00dc9f4af854 -r b22268575af1 ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Mon Mar 31 19:49:03 2025 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Mon Mar 31 19:53:58 2025 +0000
@@ -490,11 +490,11 @@
   AssertIsInMainProcess();
   AssertIsOnBackgroundThread();
 
-  auto actor = MakeRefPtr<dom::notification::NotificationParent>(
-      aPrincipal, aEffectiveStoragePrincipal, aIsSecureContext, aScope,
-      aNotification);
-  nsresult rv =
-      actor->BindToMainThread(std::move(aParentEndpoint), std::move(aResolver));
+  dom::notification::NotificationParentArgs args{
+      aPrincipal, aEffectiveStoragePrincipal, aIsSecureContext,
+      nsString(aScope), aNotification};
+  nsresult rv = dom::notification::NotificationParent::CreateOnMainThread(
+      std::move(args), std::move(aParentEndpoint), std::move(aResolver));
   if (NS_FAILED(rv)) {
     return IPC_FAIL(this, "Invalid NotificationParent constructor arguments");
   }
